guardian/admin.py

```diff
@@ -428,6 +428,66 @@ class AuthorAdmin(GuardedModelAdmin):
         admin.site.register(Author, AuthorAdmin)
 
     """
+    def get_model_objs(self, request, action=None, klass=None):
+        opts = self.opts
+        actions = [action] if action else ['view', 'change', 'delete']
+        klass = klass if klass else opts.model
+        model_name = klass._meta.model_name
+        return get_objects_for_user(user=request.user, perms=['{}_{}'.format(perm, model_name) for perm in actions],
+                                    klass=klass, any_perm=True)
+
+    def has_module_permission(self, request):
+        if super(GuardedModelAdmin, self).has_module_permission(request):
+            return True
+        return self.get_model_objs(request).exists()
+
+    def get_queryset(self, request):
+        if request.user.is_superuser:
```

Comment:
Again, moving  logic of your application to our library. Is there a shortcut to this?

Reply:
I get the impression that transferring logic, because in django-guardian, if there is a possibility, we do not abuse the superuser, and in fact we use permissions. Independently, I see the potential for change in this.

Reply:
I think this is no problem. The superuser have the all permission, and "get_queryset" return all objs for superuser is correct.